lib/doorkeeper/request.rb

```diff
@@ -9,22 +9,36 @@ module Doorkeeper
   module Request
     module_function
 
-    def authorization_strategy(strategy)
-      get_strategy strategy, Doorkeeper.configuration.authorization_response_types
-    rescue NameError
-      raise Errors::InvalidAuthorizationStrategy
+    def authorization_strategy(response_type)
+      fail Errors::MissingRequestStrategy unless response_type.present?
+
+      flow = authorization_flows.find{|f| f.matches_response_type?(response_type) }
```

Comment:
Line is too long. [83/80]<br>Prefer `detect` over `find`.<br>Space missing to the left of {.<br>Space between { and | missing.